src/Giraffe/Middleware.fs

```diff
@@ -96,7 +97,26 @@ type IApplicationBuilder with
     /// Returns `unit`.
     ///
     member this.UseGiraffe (handler : HttpHandler) =
-        this.UseMiddleware<GiraffeMiddleware> handler
+        this.UseMiddleware<GiraffeMiddleware> (handler, Version36)
```

Comment:
The default is set to `Version36` if no compatibility mode is set.

Reply:
Very nice 😄 
My only thought is that it could be that we change `UseGiraffe` to return an `IGiraffeOptionsBuilder` and then have an extension method `SetCompatibilityVersion`.

The risk is in its current state is that over time we gather up more parameters and it gets a little messy.

Or we could wrap this in a class (not a record as they don't version well).

Reply:
Fair point

Reply:
I slightly tend more towards the `GiraffeOptions` version, becaues then the `UseGiraffe` extension method doesn't change the return type and it will continue to work in a chain of middleware constelations for existing apps and it will save me to create yet another extension method on the `IGiraffeOptionsBuilder` to `Build()` the middleware and then return an `IApplicationsBuilder`. Any thoughts/tips/reasons why the `IGiraffeOptionsBuilder` might still be better?

Reply:
That's a good point, I was just straight up copying MVC 😆 
I don't think you'd expose a `Build` method, you'd do it like `IMvcBuilder`, which has a `Services` property to get back to a fluent API. So the breaking change would be:
```
service.UseGiraffe(app).UseSomeOtherThing();
```
to:
```
service.UseGiraffe(app).Services.UseSomeOtherThing();
```
which is consistent with MVC, Identity Server Authentication, HttpClientFactory and others. With `GiraffeOptions` I haven't given much thought to how it would be to consume.

Reply:
I'm just going through the MVC code to see how they've done it. Can you maybe give me some direction where to find the relevant code?

These are the questions which I'm trying to work out:

- The object which will hold the options for Giraffe, is it supposed to get registered as a dependency via the services collection? I suppose that would be the correct place as it might be used beyond just the middleware, right?
- When passing settings through an `IApplicationBuilder` extension method like `SetCompatibilityMode(...)` can I then register that object in the `IServiceCollection` too or would I need to set it a second time in `AddGiraffe()`?

Reply:
Ok, I've made another change, see below for how it works.